samza-api/src/main/java/org/apache/samza/runtime/ProcessorIdGenerator.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.runtime;
+
+import org.apache.samza.annotation.InterfaceStability;
+import org.apache.samza.config.Config;
+
+@InterfaceStability.Evolving
+public interface ProcessorIdGenerator {
+  /**
+   * Generates a String representation to identify a single instance of StreamProcessor
+   * This value can be representative of its current executing environment. It can also be custom-managed by the user.
```

Comment:
Will fix the typos

> Not sure if this is helpful: This value can be representative of its current executing environment. It can also be custom-managed by the user. I suspect it can be removed. You are already stating out the general contract for the ID. (in the following lines).

Why is not helpful though? If you think it is misleading, I can remove it. I will be updating this doc again based on some of @nickpan47 's feedback.